File: src/main/java/de/mikeyllp/miniGamesV4/games/hideandseek/listeners/PlayerHoldItemListener.java
package de.mikeyllp.miniGamesV4.games.hideandseek.listeners;

import org.bukkit.Material;
import org.bukkit.attribute.Attribute;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerItemHeldEvent;
import org.bukkit.inventory.ItemStack;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.List;
import java.util.Map;

import static de.mikeyllp.miniGamesV4.games.hideandseek.storage.HideAndSeekGameGroups.seekerGroup;

public class PlayerHoldItemListener implements Listener {

    private final JavaPlugin plugin;

    public PlayerHoldItemListener(JavaPlugin plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onPlayerHoldItem(PlayerItemHeldEvent event) {
        FileConfiguration config = plugin.getConfig();
        if (!config.getBoolean("small-modus.is-enabled")) return;
        if (seekerGroup.isEmpty()) return;

        // Checks if the player is a Seeker
        Player player = event.getPlayer();
        for (Map.Entry<String, List<Player>> entry : seekerGroup.entrySet()) {
            List<Player> seekers = entry.getValue();
            if (!seekers.contains(player)) return;
        }

        // If the player hold a puffer fish he will get smaller
        ItemStack hand = player.getInventory().getItem(event.getNewSlot());
        if (hand != null && hand.getType() == Material.PUFFERFISH) {
            player.getAttribute(Attribute.SCALE).setBaseValue(0.001);
        } else {
            player.getAttribute(Attribute.SCALE).setBaseValue(1.0);
        }


    }
}

File: src/main/java/de/mikeyllp/miniGamesV4/games/hideandseek/listeners/PlayerHoldItemListener.kt
package de.mikeyllp.miniGamesV4.games.hideandseek.listeners

import de.mikeyllp.miniGamesV4.games.hideandseek.storage.HideAndSeekGameGroups
import de.mikeyllp.miniGamesV4.plugin
import org.bukkit.Material
import org.bukkit.attribute.Attribute
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.ClickType
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.player.PlayerDropItemEvent
import org.bukkit.event.player.PlayerItemConsumeEvent
import org.bukkit.event.player.PlayerItemHeldEvent
import org.bukkit.event.player.PlayerSwapHandItemsEvent

class PlayerHoldItemListener() : Listener {

    @EventHandler
    fun onPlayerHoldItem(event: PlayerItemHeldEvent) {
        val config = plugin.config

        if (!config.getBoolean("small-modus.is-enabled")) return
        if (HideAndSeekGameGroups.seekerGroup.isEmpty()) return

        val player = event.player
        for (entry in HideAndSeekGameGroups.seekerGroup.entries) {
            val seekers: MutableList<Player> = entry.value
            if (!seekers.contains(player)) return
        }

        val hand = player.inventory.getItem(event.newSlot)
        if (hand != null && hand.type == Material.PUFFERFISH) {
            player.getAttribute(Attribute.SCALE)?.baseValue = 0.001
        } else {
            player.getAttribute(Attribute.SCALE)?.baseValue = 1.0
        }
    }

    @EventHandler
    fun onTrySwitchItem(event: PlayerSwapHandItemsEvent) {
        val config = plugin.getConfig()
        if (!config.getBoolean("small-modus.is-enabled")) return
        if (HideAndSeekGameGroups.seekerGroup.isEmpty()) return

        val player = event.getPlayer()
        for (entry in HideAndSeekGameGroups.seekerGroup.entries) {
            val seekers: MutableList<Player> = entry.value
            if (!seekers.contains(player)) return
        }

        if (event.offHandItem.type == Material.PUFFERFISH) {
            event.isCancelled = true
        }
    }

    @EventHandler
    fun onInventoryClick(event: InventoryClickEvent) {
        val config = plugin.getConfig()
        if (!config.getBoolean("small-modus.is-enabled")) return
        if (HideAndSeekGameGroups.seekerGroup.isEmpty()) return

        val player = event.whoClicked as Player
        for (entry in HideAndSeekGameGroups.seekerGroup.entries) {
            val seekers: MutableList<Player> = entry.value
            if (!seekers.contains(player)) return
        }

        if (event.click == ClickType.NUMBER_KEY && (event.currentItem?.type != Material.PUFFERFISH)) {
            event.isCancelled = true
            return
        }

        val item = event.getCurrentItem()
        if (item != null && item.type == Material.PUFFERFISH) {
            event.isCancelled = true
        }
    }

    @EventHandler
    fun onItemDrop(event: PlayerDropItemEvent) {
        for (entry in HideAndSeekGameGroups.seekerGroup.entries) {
            val seekers: MutableList<Player> = entry.value
            if (!seekers.contains(event.player)) return
        }

        if (event.itemDrop.itemStack.type == Material.PUFFERFISH) {
            event.isCancelled = true
        }
    }

    @EventHandler
    fun onItemEat(event: PlayerItemConsumeEvent) {
        for (entry in HideAndSeekGameGroups.seekerGroup.entries) {
            val seekers: MutableList<Player> = entry.value
            if (!seekers.contains(event.player)) return
        }

        if (event.item.type == Material.PUFFERFISH) {
            event.isCancelled = true
        }
    }
}
